src/main/java/at/hannibal2/skyhanni/features/skillprogress/SkillProgress.kt

```diff
@@ -0,0 +1,471 @@
+                Renderable.clickAndHover(
+                    "§cOpen your skills menu !",
+                    listOf("§eClick here to execute §6/skills"),
+                    onClick = { LorenzUtils.sendCommandToServer("skills") }
+                )
+            } else {
+                val tips = buildList {
+                    add("§6Level: §b${level}")
+                    add("§6Current XP: §b${currentXp.addSeparators()}")
+                    add("§6Needed XP: §b${currentXpMax.addSeparators()}")
+                    add("§6Total XP: §b${totalXp.addSeparators()}")
+                }
+                val nameColor = if (skill == activeSkill) "§e" else "§6"
+                Renderable.hoverTips(buildString {
+                    append("$nameColor${skill.displayName} $level ")
+                    append("§7(")
+                    append("§b${currentXp.addSeparators()}")
+                    if (currentXpMax != 0L) {
+                        append("§6/")
+                        append("§b${currentXpMax.addSeparators()}")
+                    }
+                    append("§7)")
+                }, tips)
+            }
+        }
+    }
+
+    private fun drawETADisplay() = buildList {
+        val activeSkill = activeSkill ?: return@buildList
+        val skillInfo = SkillAPI.storage?.get(activeSkill) ?: return@buildList
+        val xpInfo = skillXPInfoMap[activeSkill] ?: return@buildList
+        val skillInfoLast = oldSkillInfoMap[activeSkill] ?: return@buildList
+        oldSkillInfoMap[activeSkill] = skillInfo
+        val level = if (config.overflowConfig.enableInEtaDisplay.get() || config.customGoalConfig.enableInETADisplay) skillInfo.overflowLevel else skillInfo.level
+
+        val useCustomGoalLevel = skillInfo.customGoalLevel != 0 && skillInfo.customGoalLevel > skillInfo.overflowLevel && customGoalConfig.enableInETADisplay
+        var targetLevel = if (useCustomGoalLevel) skillInfo.customGoalLevel else level + 1
+        if (targetLevel <= level || targetLevel > 400) targetLevel = (level + 1)
+        val currentLevelNeededXp = SkillUtil.xpRequiredForLevel(level.toDouble()) + skillInfo.overflowCurrentXp
+        val targetNeededXp = SkillUtil.xpRequiredForLevel(targetLevel.toDouble())
+        var remaining = if (useCustomGoalLevel) targetNeededXp - currentLevelNeededXp else skillInfo.overflowCurrentXpMax - skillInfo.overflowCurrentXp
+
+        if (!useCustomGoalLevel) {
+            if (skillInfo.overflowCurrentXpMax == skillInfoLast.overflowCurrentXpMax) {
+                remaining = interpolate(remaining.toFloat(), (skillInfoLast.overflowCurrentXpMax - skillInfoLast.overflowCurrentXp).toFloat(), lastGainUpdate.toMillis()).toLong()
+            }
+        }
+
+        add(Renderable.string(buildString {
+            append("§6Skill: §7${activeSkill.displayName} ")
+            if (useCustomGoalLevel && targetLevel != 0) append("§8$level➜§3${targetLevel}") else append(level)
```

Comment:
With custom skill goal enabled, the eta display shows the "from -> to" line perfectly. Maybe show this format even without custom goal level enabled, and also change the color in the crop milestone display to be consistent.
![grafik](https://github.com/hannibal002/SkyHanni/assets/24389977/4cdc55ac-77e6-4f2c-ae3c-fd5c8e6b7c89)